agent/agent.go

```diff
@@ -1082,6 +1083,7 @@ func (a *agent) handleReconnectingPTY(ctx context.Context, logger slog.Logger, m
 
 		if err = a.trackConnGoroutine(func() {
 			rpty.Wait()
+			cleanup()
 			a.reconnectingPTYs.Delete(msg.ID)
 		}); err != nil {
 			rpty.Close(err)
```

Comment:
We should cleanup here too.
```suggestion
			rpty.Close(err)
			cleanup()
```